src/sagemaker_containers/transformers.py

```diff
@@ -0,0 +1,167 @@
+
+         Returns:
+             (obj): data ready for prediction.
+        """
+        raise NotImplementedError()
+
+    @staticmethod
+    def output_fn(prediction, accept):
+        """Function responsible to serialize the prediction for the response.
+
+        Args:
+            prediction (obj): prediction returned by predict_fn .
+            accept (str): accept content-type expected by the client.
+
+        Returns:
+            (TransformSpec): a namedtuple with the following args:
+
+                * Args:
+                    response: the serialized data to return
+                    accept: the content-type that the data was transformed to.
+        """
+        default_encoder = encoder.DefaultEncoder()
+        return worker.Response(default_encoder.encode(prediction, accept), accept)
+
+    def transform_fn(self, model, input_data, content_type, accept):
+        """Function responsible for input processing, prediction, and output processing.
+
+        Args:
+            model (obj): model loaded by model_fn
+            input_data (obj): the request data content.
+            content_type (str): the request Content-Type.
+            accept: the content-type that the data was transformed to.
+
+        Returns:
+            (TransformSpec): a namedtuple with the following args:
+
+                * Args:
+                    response: the serialized data to return
+                    accept: the content-type that the data was transformed to.
+        """
+        data = self.input_fn(input_data=input_data, content_type=content_type)
+        prediction = self.predict_fn(model=model, data=data)
+        return self.output_fn(prediction=prediction, accept=accept)
+
+    def initialize(self):  # type: () -> None
+        """Execute any initialization necessary to start making predictions with the Transformer.
+
+        The default implementation is used to load the model.
+
+        This function is called by sagemaker_containers.worker.Worker, before starting the Flask application.
+        The gunicorn server forks multiple workers, executing multiple Flask applications in parallel.
+        This function will be called once per each worker.
+
+        It does not have return type or arguments.
+        """
+        self._model = self.model_fn(model_dir=env.ServingEnv().model_dir)
+
+    def transform(self):  # type: () -> worker.Response
+        """Responsible to make predictions against the model.
+
+        Returns:
+            (sagemaker_containers.transformers.TransformSpec): named tuple with prediction data.
+        """
+        request = worker.Request()
+
+        result = self.transform_fn(self._model, request.content, request.content_type, request.accept)
+
+        if not isinstance(result, worker.Response):
+            # transforms tuple in Response for backwards compatibility
+            return worker.Response(response=result[0], accept=result[1])
+
+        return result
```

Comment:
Now that the Worker is a Flask application, this transform is basically a view functions: It creates the request object, makes the prediction and returns a Response object.